packages/web/src/components/typography/Text.tsx

```diff
@@ -0,0 +1,87 @@
+// import cn from 'classnames'
+
+import { CSSProperties, ReactNode } from 'react'
+
+import { getCurrentThemeColors, ThemeColor } from 'utils/theme/theme'
+
+import { variantInfoMap, variantStrengthMap } from './constants'
+import {
+  TextSizeInfo,
+  TextStrengthInfo,
+  TextVariant,
+  TextSize,
+  TextStrength,
+  FontWeight
+} from './types'
+
+export const fontWeightMap: Record<FontWeight, number> = {
+  heavy: 900,
+  bold: 700,
+  demiBold: 600,
+  medium: 500,
+  regular: 400,
+  light: 300,
+  thin: 200,
+  ultraLight: 100
+}
+
+const defaultBodyInfo: TextSizeInfo & TextStrengthInfo = {
+  tag: 'p',
+  fontSize: '16px',
+  lineHeight: '20px',
+  fontWeight: 'medium',
+  textTransform: 'inherit'
+}
+
+export type TextProps = {
+  className?: string
+  children?: ReactNode
+  variant?: TextVariant
+  size?: TextSize
+  strength?: TextStrength
+  color?: ThemeColor
+}
+
+export const Text = ({
+  className,
+  children,
+  variant = 'body',
+  strength = 'default',
+  size = 'M',
+  color = '--neutral',
+  ...props
+}: TextProps) => {
+  const themeColors = getCurrentThemeColors()
+  const textColor = themeColors[color] ?? themeColors['--neutral']
+
+  const variantSizeInfo = variantInfoMap[variant][size] ?? {}
+  const variantStrengthInfo = variantStrengthMap[variant][strength] ?? {}
+
+  const {
+    tag: Tag,
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    fontWeight,
+    textTransform
+  } = {
+    ...defaultBodyInfo,
+    ...variantSizeInfo,
+    ...variantStrengthInfo
+  }
+
+  const styleObject: CSSProperties = {
```

Comment:
as a first implementation, using styleObject is okay, but long term i do think it makes sense to use classNames so we can get better performance and readability when inspecting elements

Reply:
yea you're probably right here. using the styles like this will also make overwriting the styles more annoying. I'll update it in the future